runtime/lua/vim/lsp/buf.lua

```diff
@@ -23,6 +23,13 @@ local function client_positional_params(params)
   end
 end
 
+--- @param callback? function
+local function invoke_callback(callback, ...)
+  if callback then
+    pcall(callback, ...)
```

Comment:
pcall masks errors. If you don't want errors to potentially mess up the flow then you can invoke the callback in `vim.schedule`